runtime/executor/src/main/java/org/apache/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -377,6 +450,22 @@ private void doExecute() {
     }
   }
 
+  /**
+   * Send data-processing metrics.
+   */
+  public void sendMetrics() {
+    metricMessageSender.send(TASK_METRIC_ID, taskId, "boundedSourceReadTime",
```

Comment:
Maybe we need to create Metrics class and add static variables, instead of using hard-coded strings.